### ISSUE TITLE
Upgrade from C124 -> C125 on Android ARM x64 via 1.67.x - Nightly (Checking C125)

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#38041]
As per process, QA runs through the following cases to ensure that the major chromium bump that's about to get merged into `master` doesn't regress the `Nightly` channel. Once the major chromium bump is merged into `master`, QA will run through a full manual pass on `Windows` & `Android`.

### Startup & Components

- [x] Verify that Brave is only contacting `*.brave.com` endpoints on first launch using either `Charles Proxy`, `Fiddler`, `Wireshark` or `LittleSnitch` (or a similar application)
  - [x] Verify that opening a NTP doesn't trigger any outbound connections related to widgets without user interaction
- [x] Restart the browser, load `brave://components`, wait for 8 mins and verify that no component shows any errors

**Note:** Always double check `brave://components` to make sure there's no errors/missing version numbers. We basically want to make sure that components are working as they regressed in C98 when we first merged into `master`.

### Upgrade

- [x] Make sure that data from the last version appears in the new version OK
- [x] Ensure that `brave://version` lists the expected Brave & Chromium versions
- [x] With data from the last version, verify that:
  - [x] Bookmarks on the bookmark toolbar and bookmark folders can be opened
  - [x] Cookies are preserved
  - [ ] Installed extensions are retained and work correctly
  - [x] Opened tabs can be reloaded
  - [x] Stored passwords are preserved
  - [x] Sync chain created in previous version is retained
  - [x] Social media-blocking buttons changes are retained
  - [x] Rewards
    - [x] Wallet balance is retained
    - [x] Auto-contribute list is retained
    - [x] Both Tips and Monthly Contributions are retained
    - [x] Wallet panel transactions list is retained
    - [x] Changes to rewards settings are retained
    - [x] Ensure that Auto Contribute is not being enabled when upgrading to a new version if AC was disabled
  - [x] Ads
    - [x] Both `Estimated pending rewards` & `Ad notifications received this month` are retained
    - [x] Changes to ads settings are retained
    - [x] Ensure that ads are not being enabled when upgrading to a new version if they were disabled
    - [x] Ensure that ads are not disabled when upgrading to a new version if they were enabled

**Comment by Uni-verse** [comment]
Verified on `Samsung Galaxy S21` using version: 

```
Brave	1.67.59 Chromium: 125.0.6422.26 (Official Build) canary (64-bit) 
Revision	96b3194901e5b7509e4e2d7408e73b67c552aa77
OS	Android 13; Build/TP1A.220624.014; 33; REL
```
Example | Example | Example
-|-|-
![Screenshot 2024-05-03 at 10 06 46 AM](https://github.com/brave/brave-browser/assets/17885425/e57928d4-dd9c-4729-97e3-319e38204d8e) | ![Screenshot 2024-05-03 at 10 07 03 AM](https://github.com/brave/brave-browser/assets/17885425/1acaa621-98ff-42bb-9bdf-dcb0e64e50e7) | ![Screenshot 2024-05-03 at 10 15 38 AM](https://github.com/brave/brave-browser/assets/17885425/5d1181ac-5e5b-43aa-8ec1-2c4784a3382e)
